### PR TITLE
Add Hikari tests for the read-write splitting plugin

### PR DESCRIPTION
**Pull request by congoamz**
### Summary

Add Hikari tests for the read-write splitting plugin

### Description

Also added read-write splitting tests for the following connection strings:
- Initial connection to reader cluster
- Initial connection to reader instance via IP address  

### Additional Reviewers

@karenc-bq 
@sergiyv-bitquill

### CODE REVIEW

File: src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class HikariCPReadWriteSplittingTest extends AuroraMysqlIntegrationBaseTest {
+
+    private static Log log = null;
+    private static final String URL_SUFFIX = PROXIED_DOMAIN_NAME_SUFFIX + ":" + MYSQL_PROXY_PORT;
+    private static HikariDataSource data_source = null;
```
Comment (karenc-bq) [inline src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java:68]:
```suggestion
    private static HikariDataSource dataSource = null;
```

File: src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+        config.addDataSourceProperty(PropertyKey.socketTimeout.getKeyName(), "3000");
+
+        return config;
+    }
+
+    private static void addAllPlugins(HikariConfig config) {
+        config.addDataSourceProperty(PropertyKey.connectionPluginFactories.getKeyName(),
+                "com.mysql.cj.jdbc.ha.plugins.ReadWriteSplittingPluginFactory," +
+                        "com.mysql.cj.jdbc.ha.plugins.failover.FailoverConnectionPluginFactory," +
+                        "com.mysql.cj.jdbc.ha.plugins.NodeMonitoringConnectionPluginFactory");
```
Comment (karenc-bq) [inline src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java:441]: The indentation looks a bit off here

File: src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java (hunk 1 above)
Comment (congoamz) [reply src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java]: Thanks, fixed now

File: src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java (hunk 2 above)
Comment (congoamz) [reply src/test/java/testsuite/integration/container/aurora/HikariCPReadWriteSplittingTest.java]: Thanks, fixed now

File: src/test/java/testsuite/integration/container/aurora/AuroraMysqlIntegrationBaseTest.java [hunk 3]
```diff
@@ -323,7 +323,7 @@ protected void assertFirstQueryThrows(Statement stmt, String expectedSQLErrorCod
     assertEquals(expectedSQLErrorCode, exception.getSQLState());
   }
 
-  protected Connection createPooledConnectionWithInstanceId(String instanceID) throws SQLException {
+  protected BasicDataSource createPooledConnectionWithInstanceId(String instanceID) throws SQLException {
```
Comment (sergiyvamz) [inline src/test/java/testsuite/integration/container/aurora/AuroraMysqlIntegrationBaseTest.java:326]: createDataSourceWithInstanceId?

**Review by sergiyvamz — APPROVED** [review]

**Pull request merged**